stdlib/float.mli

```diff
@@ -257,11 +257,264 @@ val hash: t -> int
+      function is applied with the index of the element as first argument,
+      and the element itself as second argument. *)
+
+  val map : (float -> float) -> t -> t
+  (** [map f a] applies function [f] to all the elements of [a],
+      and builds a floatarray with the results returned by [f]. *)
+
+  val mapi : (int -> float -> float) -> t -> t
+  (** Same as {!map}, but the
+      function is applied to the index of the element as first argument,
+      and the element itself as second argument. *)
+
+  val fold_left : ('a -> float -> 'a) -> 'a -> t -> 'a
+  (** [fold_left f x a] computes
+      [f (... (f (f x a.(0)) a.(1)) ...) a.(n-1)],
+      where [n] is the length of the floatarray [a]. *)
+
+  val fold_right : (float -> 'a -> 'a) -> t -> 'a -> 'a
+  (** [fold_right f a x] computes
+      [f a.(0) (f a.(1) ( ... (f a.(n-1) x) ...))],
+      where [n] is the length of the floatarray [a]. *)
+
+  val iter2 : (float -> float -> unit) -> t -> t -> unit
+  (** [Array.iter2 f a b] applies function [f] to all the elements of [a]
+      and [b].
+      Raise [Invalid_argument] if the floatarrays are not the same size. *)
+
+  val map2 : (float -> float -> float) -> t -> t -> t
+  (** [map2 f a b] applies function [f] to all the elements of [a]
+      and [b], and builds a floatarray with the results returned by [f]:
+      [[| f a.(0) b.(0); ...; f a.(length a - 1) b.(length b - 1)|]].
+      Raise [Invalid_argument] if the floatarrays are not the same size. *)
+
+  val for_all : (float -> bool) -> t -> bool
+  (** [for_all p [|a1; ...; an|]] checks if all elements of the floatarray
+      satisfy the predicate [p]. That is, it returns
+      [(p a1) && (p a2) && ... && (p an)]. *)
+
+  val exists : (float -> bool) -> t -> bool
+  (** [exists p [|a1; ...; an|]] checks if at least one element of
+      the floatarray satisfies the predicate [p]. That is, it returns
+      [(p a1) || (p a2) || ... || (p an)]. *)
+
+  val mem : float -> t -> bool
+  (** [mem a l] is true if and only if [a] is equal (in the sense of generic
+      equality) to an element of [l]. *)
+
+  val memq : float -> t -> bool
+  (** Same as {!mem}, but uses float equality instead of generic
+      equality to compare elements. *)
+
+  val sort : (float -> float -> int) -> t -> unit
+  (** Sort a floatarray in increasing order according to a comparison
+      function.  The comparison function must return 0 if its arguments
+      compare as equal, a positive integer if the first is greater,
+      and a negative integer if the first is smaller (see below for a
+      complete specification).  For example, {!Pervasives.compare} is
+      a suitable comparison function.  After calling [sort], the
+      array is sorted in place in increasing order.
+      [sort] is guaranteed to run in constant heap space
+      and (at most) logarithmic stack space.
+
+      The current implementation uses Heap Sort.  It runs in constant
+      stack space.
+
```

Comment:
(Don't we need to discuss NaN values?)

Reply:
I think the section "specification of the comparison function" takes care of that by ruling out using the IEEE comparison operators.

Reply:
OK